crates/rspack_loader_swc/src/compiler.rs

```diff
@@ -359,21 +362,37 @@ impl SwcCompiler {
       match result {
         Ok(v) => Ok(v),
         Err(err) => {
-        let error_msg = match err.downcast_ref::<String>(){
-          Some(msg) => {
-            msg
-          },
-          None => "unknown error"
-        };
-        let swc_core_version = env!("RSPACK_SWC_CORE_VERSION");
-        // FIXME: with_help has bugs, use with_help when diagnostic print is fixed
-        let help_msg = formatdoc!{"
-          The version of the SWC Wasm plugin you're using might not be compatible with `builtin:swc-loader`.
-          The `swc_core` version of the current `rspack_core` is {swc_core_version}. 
-          Please check the `swc_core` version of SWC Wasm plugin to make sure these versions are within the compatible range.
-          See this guide as a reference for selecting SWC Wasm plugin versions: https://rspack.dev/errors/swc-plugin-version"};
-        let report: Report = MietteDiagnostic::new(format!("{}{}",error_msg,help_msg)).with_code("Builtin swc-loader error").into();
-        Err(report)
+
+        let swc_diagnostics = err.diagnostics();
+        if swc_diagnostics.iter().any(|d| match &d.code {
+            Some(code) => {
+              // reference to: 
+              //    https://github.com/swc-project/swc/blob/v1.11.21/crates/swc/src/plugin.rs#L187
+              //    https://github.com/swc-project/swc/blob/v1.11.21/crates/swc/src/plugin.rs#L200
+              match code {
+                swc_core::common::errors::DiagnosticId::Error(e) => e.contains("plugin"),
+                swc_core::common::errors::DiagnosticId::Lint(_) => false,
+              }
+            },
+            None => false,
+        }
+      ) {
+          // swc errors includes plugin error;
+          let error_msg = err.to_pretty_string();
+          let swc_core_version = env!("RSPACK_SWC_CORE_VERSION");
+          // FIXME: with_help has bugs, use with_help when diagnostic print is fixed
```

Comment:
Can you provide any context for these bugs?

Reply:
Acorrding this PR https://github.com/swc-project/swc/pull/10241, swc api try_with_handler return TWithDiagnosticArray struct, we can get the diagnostic id to judge swc error type.
Only emit help message when swc load plugin error.